.azure-pipelines/azure-pipelines-merge.yml

```diff
@@ -148,28 +148,37 @@ jobs:
 
   - template: templates/setup-ci-machine.yml
 
-  - script: 'pip install --upgrade .'
-    displayName: 'Install Azure DevOps CLI extension'
-    workingDirectory: 'azure-devops/'
+  - template: templates/download-install-local-azure-devops-cli-extension.yml
+
+  - task: PythonScript@0
+    displayName: 'Find Extension Directory'
+    inputs:
+      scriptSource: 'inline'
+      script: |
+        from azure.cli.core.extension import get_extension_path
+        extensionPath = get_extension_path('azure-devops')
+        print("##vso[task.setvariable variable=extensionPath]"+extensionPath)
+      failOnStderr: true
```

Comment:
A lot of inline script code we are adding in yml is making the pipelines complex. 

1. Whatever we need in the pipeline will also be required by the developer so it makes sense to add these scripts in the code repo itself.
2. Since the pipelines are harder to debug, using locally tested scripts in the pipelines will keep the pipelines up-to-date and more robust.